### PR TITLE
Fix #21: add support for task lists

### LINKED ISSUE #21: Checkbox parsing
See https://github.com/GitbookIO/feedback/issues/116

### PR DESCRIPTION
**Pull request by SamyPesse**
The goal of this PR is to fix #21 and GitbookIO/feedback#116.

GitHub supports task lists since [2013](https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments).

Checkboxes can only be put at the beginning of a list item. It is not a node by itself, but a property of the list item.

This PR adds parsing for these checkboxes and store the status `checked<Boolean>` as a data property of the list items (Similar to https://github.com/chjj/marked/pull/587).

- [x] Deserialize task lists from markdown
- [x] Serialize task lists to markdown
- [x] Serialize task lists to HTML, follow the same class names as GitHub
- [ ] Deserialize task lists from HTML ? This is one may be tricky but can be useful when pasting content from GitHub in the editor.

### CODE REVIEW

**Comment by SamyPesse** [comment]
@Soreine It should be ready for review.

The main issue is that it is not as strict as GitHub, for example the following snippet is not pared the same way on `markup-it` and GitHUb GFM:

```markdown
- [x] # An heading parsed ny markup-it but not GFM
```

File: src/html/blocks/listitem.js [hunk 1]
```diff
@@ -8,11 +8,20 @@ const serialize = Serializer()
...
         const node = state.peek();
-        const inner = state.serialize(node.nodes);
+        let inner = state.serialize(node.nodes);
+        let tag = '<li>';
+
+        const isTaskList = node.data.has('checked');
+        const isChecked = node.data.get('checked');
+
+        if (isTaskList) {
+            tag = '<li class="task-list-item">';
```
Comment (Soreine) [inline src/html/blocks/listitem.js]: Could you rather put `className` as a variable

`className = ' class="task-list-item"'`
or 
`className = ''`

File: src/html/blocks/list.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ])
+    .then(state => {
+        const node = state.peek();
+        const tag = node.type === BLOCKS.OL_LIST ? 'ol' : 'ul';
+        const isTaskList = containsTaskList(node);
+        const inner = state.serialize(node.nodes);
+
+        return state
+            .shift()
+            .write(`<${tag}${isTaskList ? ' class="contains-task-list"' : ''}>\n${inner}</${tag}>\n`);
```
Comment (Soreine) [inline src/html/blocks/list.js]: I don't know if the naming matters, but what do you prefer between `contains-task-list`,  `with-tasks`, and `task-list` ?

Comment (SamyPesse) [inline src/html/blocks/list.js]: It matters, it's the same class names as GitHub.

**Pull request merged**